{{cookiecutter.project_slug}}/pyproject.toml

```diff
@@ -110,8 +107,11 @@ trailing_comma_inline_array = true
 legacy_tox_ini = """
     [gh-actions]
     python =
-    {%- for version in python_versions -%}
-        version: py{{ version | replace: '.', ''}
+    {% for python_version in range(
+        cookiecutter.min_python_version | replace('3.', '') | int,
+        cookiecutter.max_python_version | replace('3.', '') | int
+    )%}
+        3.{{ python_version }}: py3{{ python_version }}
```

Comment:
at the minute this formats the list of versions like this:
```
python =

3.8: py38

3.9: py39

3.10: py310

``` 
I think it should format without newlines if you move `3.{{ python_version }}: py3{{ python_version }}` to the same line as the closing brace in the for loop

Reply:
Ah yeah didn't think to look at that. Oops

Reply:
It doesn't look like that on mine! But will see if I can remove the whitespace

Reply:
Looks like this for me
```
[tool.tox]
legacy_tox_ini = """
    [gh-actions]
    python =
    
        3.8: py38
    
        3.9: py39
    
        3.10: py310
    

    [testenv]
    commands =
        pytest --cov
    deps =
        pytest
        pytest-cov

    [tox]
    env_list =
    
        py38
    
        py39
    
        py310
    
"""
```

Reply:
for some reason you need to commit the change before you can run cookiecutter on your local folder and see the changes. I guess it still relies on git history when running locally?

Reply:
after commiting,  that section looks like this for me:
```
    [gh-actions]
    python =
     3.8: py38
     3.9: py39
     3.10: py310
    

    [testenv]
```
So maybe also remove the newline before `[testenv]` in the template

Reply:
sorry just saw you've already fixed it!

Reply:
The key is using i.e. `{%-` rather than `{%` to remove whitespace before

Reply:
ah that's good to know, thanks!